### ISSUE TITLE
AssertionError on load() when using MultiDiscrete

### ISSUE THREAD

**Issue by nagadomi** [hill-a/stable-baselines#25]
**Describe the bug**
 https://github.com/hill-a/stable-baselines/blob/70898bd8ccdedffb8a59e5013a0d2300b331c4f8/stable_baselines/common/base_class.py#L89-L90
When using `gym.spaces.MultiDiscrete` as an action, AssertionError will occur at the above line because `MultiDiscrete` does not have `__eq__` method. I think it is a problem on `gym` :(

**Code example**
```python
from gym import spaces

# pass
assert spaces.Discrete(4) == spaces.Discrete(4)
# assertion error
assert spaces.MultiDiscrete([1, 2, 3]) == spaces.MultiDiscrete([1, 2, 3])
```

**Comment by araffin** [comment]
Hello, 
Thanks for reporting the bug =)
Yes, I think you are right, this is a gym related bug.

Can you open an issue on OpenAI Gym repo? (and maybe refer to this issue)

@hill-a should we patch the code until gym is updated?

**Comment by nagadomi** [comment]
hi,
>Can you open an issue on OpenAI Gym repo? (and maybe refer to this issue)

In previous version of gym, MultiDiscrete had `__eq__` method but it seems to have been removed.  
I am not familiar with this field. So I leave it to you.

**Comment by hill-a** [comment]
Hey,

made a fix on 678f803 and d768aff, they will be merged with master in a short amount of time. 
In the mean time, you can simply add this snippet to the start of your code, that should fix this issue:

```python
import gym
import numpy as np

if gym.spaces.MultiBinary.__eq__ == object.__eq__:
    def _eq(self, other):
        return self.n == other.n

    gym.spaces.MultiBinary.__eq__ = _eq

if gym.spaces.MultiDiscrete.__eq__ == object.__eq__:
    def _eq(self, other):
        return np.all(self.nvec == other.nvec)

    gym.spaces.MultiDiscrete.__eq__ = _eq
```

As a side note, it seems that MultiBinary has the same issue.

**Comment by araffin** [comment]
I submitted the PR: https://github.com/openai/gym/pull/1178
Should be fixed in newer versions of gym.
